Implement fullscreen in GTK4 platform

MortimerGoro opened the pull request:
Fixes https://github.com/Igalia/cog/issues/287

aperezdc reviewed (changes requested):
Could we have the usage of the new libwpe API guarded with `#if WPE_CHECK_VERSION(1, 11, 1)`? We would like to allow building Cog with the last two or three stable series of the dependency libraries.

Other than that, I like this patch and it goes to show how much GTK does for us under the hood 😄

MortimerGoro commented:
> Could we have the usage of the new libwpe API guarded with `#if WPE_CHECK_VERSION(1, 11, 1)`? We would like to allow building Cog with the last two or three stable series of the dependency libraries.
> 
> Other than that, I like this patch and it goes to show how much GTK does for us under the hood smile

Thanks for the review. I already added the WPE version check guards, and fixed the style checks

aperezdc reviewed (approved):
Thanks for the update! 🎉
